### PR TITLE
Add :deploy_pip to //grpc/python

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

To make client-python easier to develop - for both us and for the community.

Currently `protocol` is built via Bazel, imported and packaged directly into `grakn-client`. This is a pain for

- non-Bazel users
- and also for us, since the `_pb2.py` files get sent to `bazel-bin` when imported into `client-python`, meaning the IDE has no idea of their existence

## What are the changes implemented in this PR?

- Add :deploy_pip to //grpc/python: the Python protocol will now be deployed to `pip`
- The resulting `pip` package will have `graknprotocol` as the root directory, allowing IDEs to resolve imports gracefully

## Example usage

The requirement on `grakn-protocol` can be specified in requirements.txt as follows:
```
--extra-index-url https://repo.grakn.ai/repository/pypi-snapshot/simple
grakn-protocol==0.0.0-07b75dc95bba6b62e0c4ee98d9859fb697653483
```
Here, `extra-index-url` is used to register our Nexus repo as an additional pip repository URL.

Worth noting that this is not ideal as we don't really want this URL to be used by the general public - however it does give us what we need for development: a reliable way of importing _any_ `grakn-protocol` snapshot into `client-python`.

### CODE REVIEW

File: .grabl/automation.yml [hunk 1]
```diff
@@ -68,6 +68,20 @@ build:
...
+      filter:
+        owner: graknlabs
+        branch: master
+      machine: graknlabs-ubuntu-20.04
+      type: foreground
+      script: |
+        pyenv global 3.6.10
+        sudo unlink /usr/bin/python3
+        sudo ln -s $(which python3) /usr/bin/python3
+        sudo ln -s /usr/share/pyshared/lsb_release.py /opt/pyenv/versions/3.6.10/lib/python3.6/site-packages/lsb_release.py
```
Comment (alexjpwalker) [inline .grabl/automation.yml:81]: These symlinks are required in order to make Python reliably work with Bazel.

File: .grabl/automation.yml [hunk 2]
```diff
@@ -68,6 +68,20 @@ build:
...
         bazel run --define version=$(git rev-parse HEAD) //grpc/nodejs:deploy-npm -- snapshot
       dependencies: [build, build-dependency]
+    deploy-pip-snapshot:
+      filter:
+        owner: graknlabs
+        branch: master
+      machine: graknlabs-ubuntu-20.04
+      type: foreground
+      script: |
+        pyenv global 3.6.10
```
Comment (alexjpwalker) [inline .grabl/automation.yml:78]: This is the latest currently installed version of Python in Grabl.

File: grpc/python/BUILD [hunk 3]
```diff
@@ -53,3 +54,38 @@ checkstyle_test(
...
+        "License :: OSI Approved :: GNU Affero General Public License v3 or later (AGPLv3+)",
+        "Operating System :: OS Independent",
+        "Intended Audience :: Developers",
+        "Intended Audience :: Science/Research",
+        "Environment :: Console",
+        "Topic :: Database :: Front-Ends"
+    ],
+    url = "https://github.com/graknlabs/protocol/",
+    author = "Grakn Labs",
+    author_email = "grabl@grakn.ai",
```
Comment (alexjpwalker) [inline grpc/python/BUILD:78]: Is this the right email address to use?

**Review by alexjpwalker — COMMENTED** [review]
These changes look good

File: grpc/python/BUILD [hunk 4]
```diff
@@ -38,13 +39,13 @@ python_grpc_compile(
 python_repackage(
     name = "protocol_pkg",
     src = ":protocol_src",
-    package = "grakn_protocol"
+    package = "graknprotocol"
```
Comment (alexjpwalker) [inline grpc/python/BUILD:42]: Nexus Repo dislikes underscores in package names. Python dislikes hyphens in package names.

File: grpc/python/BUILD [hunk 5]
```diff
@@ -38,13 +39,13 @@ python_grpc_compile(
...
 )
 
 py_library(
     name = "protocol",
     srcs = ["protocol_pkg"],
-    imports = ["../../grpc/python"]
+    imports = ["../../../grpc/python"]
```
Comment (alexjpwalker) [inline grpc/python/BUILD:48]: This should always have been the import path- without this change, the pip package was putting `graknprotocol` inside `bazel-out/darwin-fastbuild/...` which was causing the IDE to fail to resolve the import.

**Review by lolski — APPROVED** [review]

**Pull request merged**